src/background/executor.ts

```diff
@@ -155,149 +174,132 @@ function backgroundListener(
-              ...payload,
-            },
-          },
-          { frameId: target.frameId }
-        );
-      });
-    }
-    case MESSAGE_RUN_BLOCK_TARGET: {
-      const target = tabToTarget.get(sender.tab.id);
+handlers.set(MESSAGE_RUN_BLOCK_TARGET, async (request, sender) => {
+  const target = tabToTarget.get(sender.tab.id);
 
-      if (!target) {
-        return Promise.reject(new BusinessError("Sender tab has no target"));
-      }
+  if (!target) {
+    throw new BusinessError("Sender tab has no target");
+  }
 
-      console.debug(`Waiting for target tab ${target} to be ready`);
-      // for now, only support top-level frame as target
-      return waitReady({ tabId: target, frameId: 0 }).then(async () => {
-        console.debug(
-          `Sending ${CONTENT_MESSAGE_RUN_BLOCK} to target tab ${target} (sender=${sender.tab.id})`
-        );
-        return browser.tabs.sendMessage(
-          target,
-          {
-            type: CONTENT_MESSAGE_RUN_BLOCK,
-            payload: {
-              sourceTabId: sender.tab.id,
-              ...request.payload,
-            },
-          },
-          { frameId: 0 }
-        );
-      });
-    }
-    case MESSAGE_ACTIVATE_TAB: {
-      return (
-        browser.tabs
-          .update(sender.tab.id, {
-            active: true,
-          })
-          // ignore the returned tab
-          .then(() => {})
-      );
-    }
-    case MESSAGE_CLOSE_TAB: {
-      return new Promise<unknown>((resolve) => {
-        browser.tabs.remove(sender.tab.id).then(resolve);
-      });
-    }
-    case MESSAGE_OPEN_TAB: {
-      return browser.tabs
-        .create(request.payload as Tabs.CreateCreatePropertiesType)
-        .then((tab) => {
-          // FIXME: include frame information here
-          tabToTarget.set(sender.tab.id, tab.id);
-          tabToOpener.set(tab.id, sender.tab.id);
-        });
-    }
-    case MESSAGE_CONTENT_SCRIPT_READY: {
-      const tabId = sender.tab.id;
-      const { frameId } = sender;
-      console.debug(`Marked tab ${tabId} (frame: ${frameId}) as ready`, {
-        sender,
-      });
+  console.debug(`Waiting for target tab ${target} to be ready`);
+  // for now, only support top-level frame as target
+  await waitReady({ tabId: target, frameId: 0 });
+  console.debug(
+    `Sending ${CONTENT_MESSAGE_RUN_BLOCK} to target tab ${target} (sender=${sender.tab.id})`
+  );
+  return browser.tabs.sendMessage(
+    target,
+    {
+      type: CONTENT_MESSAGE_RUN_BLOCK,
+      payload: {
+        sourceTabId: sender.tab.id,
+        ...request.payload,
+      },
+    },
+    { frameId: 0 }
+  );
+});
 
-      const url = new URL(sender.url);
-      const nonce = url.searchParams.get("_pb");
+handlers.set(MESSAGE_ACTIVATE_TAB, async (request, sender) => {
+  await browser.tabs.update(sender.tab.id, {
+    active: true,
+  });
+});
+
+handlers.set(MESSAGE_CLOSE_TAB, async (request, sender) => {
+  return browser.tabs.remove(sender.tab.id);
```

Comment:
This is one of those cases where I did not change the original behavior, but can be safely dropped because `.remove()` returns `Promise<void>`
```suggestion
  await browser.tabs.remove(sender.tab.id);
```